ZuluContent/Zulu/Engines/Loot/LootGenerator.cs

```diff
@@ -177,15 +192,32 @@ protected override void OnTarget(Mobile from, object targeted)
         public static void MakeLoot(Container container, LootTable table)
         {
             var lootItems = table.Roll();
+            var itemsDict = new Dictionary<string, Item>();
```

Comment:
You probably don't need a dictionary here, you just need to be able to group the stackables together so you can sum their quantity/amounts.

As an example this is how you could do it using LINQ

```csharp
public static void MakeLoot(Container container, LootTable table)
{
    var items = table.Roll()
        .Select(lootItem =>
        {
            // Attempt to make the item magical, will do nothing if its not eligible.
            MakeItemMagical(lootItem); 
            return lootItem.Create();
        })
        // Group the items by id and whether they're stackable
        .GroupBy(item => new {item.Stackable, item.ItemID}) 
        // Flattens a collection containing many collections of the same type,
        // i.e. List<List<Item>> becomes List<Item>
        .SelectMany(group => 
        {
            // Merge the stackable items into the first instance and return that as an array to be flattened
            if (group.Key.Stackable) 
            {
                var item = group.First();
                item.Amount = group.Sum(x => x.Amount);
                return new[] {item};
            }
            
            // If this group of items is not stackable just return the entire group
            return group.ToArray();
        });

    foreach (var item in items)
        container.AddItem(item);
}
```

Reply:
Sweet, that works really nicely implemented it